src/Entity/EntityId.php

```diff
@@ -7,6 +7,38 @@
 use Serializable;
 
 /**
+ * Base class for value objects representing the IDs of entities.
+ * Multiple "equivalent" EntityIds may refer to the same Entity without
+ * being equal.
+ *
+ * The EntityId class is abstract, since any concrete entity ID should be
+ * represented by a subclass specific to the respective type of entity.
+ * This leaves each entity type free to define its own identifier syntax.
+ * EntityId objects are tightly bound to their string representation
+ * as returned by the getSerialization method.
+ *
+ * Entity IDs can specify the name of a context (a Wikibase repository) in
+ * which they can be resolved. IDs that do not provide a repository name
+ * (or, technically, have the repository name set to the empty string, '')
+ * are referred to as "local". IDs that do provide a repository name are
+ * referred to as "foreign" (even if they, by some detour, are equivalent to
+ * a local ID).
+ *
+ * The intended interpretation of "local" IDs is that they can be resolved
+ * in the default context - that is, they belong to the Wikibase repository
+ * the EntityId object was instantiated in, or, for Wikibase  clients,
+ * the default repo of the client the EntityId object was instantiated in.
```

Comment:
This is confusing and I believe even wrong. What this sentence basically says is that the interpretation of "local" is meaningless. Could mean anything you want. May not even be "local". Also it's unclear to what "instantiated" refers to if an ID is, for example, read from a database or dump. Same for "default". To what does this refer to? Who defines said "default"? Who knows about it? Certainly not the DataModel.